packages/cli/src/LoadNodesAndCredentials.ts

```diff
@@ -304,279 +245,155 @@ class LoadNodesAndCredentialsClass {
-		};
-
-		// eslint-disable-next-line consistent-return
-		return {
-			name: fullNodeName,
-			version: nodeVersion,
-		} as INodeTypeNameVersion;
+	private unloadNodes(installedNodes: InstalledNodes[]): void {
+		installedNodes.forEach((installedNode) => {
+			delete this.loaded.nodes[installedNode.type];
+		});
 	}
 
-	/**
-	 * Retrieves `categories`, `subcategories`, partial `resources` and
-	 * alias (if defined) from the codex data for the node at the given file path.
-	 *
-	 * @param {string} filePath The file path to a `*.node.js` file
-	 */
-	getCodex(filePath: string): CodexData {
-		// eslint-disable-next-line global-require, import/no-dynamic-require, @typescript-eslint/no-var-requires
-		const { categories, subcategories, resources: allResources, alias } = require(`${filePath}on`); // .js to .json
-
-		const resources = pick(allResources, ['primaryDocumentation', 'credentialDocumentation']);
-
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-return
-		return {
-			...(categories && { categories }),
-			...(subcategories && { subcategories }),
-			...(resources && { resources }),
-			...(alias && { alias }),
-		};
+	private attachNodesToNodeTypes(installedNodes: InstalledNodes[]): void {
+		const loadedNodes = this.loaded.nodes;
+		installedNodes.forEach((installedNode) => {
+			const { type, sourcePath } = loadedNodes[installedNode.type];
+			loadedNodes[installedNode.type] = { type, sourcePath };
+		});
 	}
 
 	/**
-	 * Adds a node codex `categories` and `subcategories` (if defined)
-	 * to a node description `codex` property.
-	 *
-	 * @param obj.node Node to add categories to
-	 * @param obj.filePath Path to the built node
-	 * @param obj.isCustom Whether the node is custom
+	 * Run a loader of source files of nodes and credentials in a directory.
 	 */
-	addCodex({
-		node,
-		filePath,
-		isCustom,
-	}: {
-		node: INodeType | IVersionedNodeType;
-		filePath: string;
-		isCustom: boolean;
-	}) {
-		try {
-			const codex = this.getCodex(filePath);
-
-			if (isCustom) {
-				codex.categories = codex.categories
-					? codex.categories.concat(CUSTOM_NODES_CATEGORY)
-					: [CUSTOM_NODES_CATEGORY];
-			}
-
-			node.description.codex = codex;
-		} catch (_) {
-			this.logger.debug(`No codex available for: ${filePath.split('/').pop() ?? ''}`);
-
-			if (isCustom) {
-				node.description.codex = {
-					categories: [CUSTOM_NODES_CATEGORY],
-				};
+	private async runDirectoryLoader<T extends DirectoryLoader>(
+		constructor: new (...args: ConstructorParameters<typeof DirectoryLoader>) => T,
+		dir: string,
+	) {
+		const loader = new constructor(dir, this.excludeNodes, this.includeNodes);
+		await loader.loadAll();
+
+		// list of node & credential types that will be sent to the frontend
+		const { types } = loader;
+		this.types.nodes = this.types.nodes.concat(types.nodes);
+		this.types.credentials = this.types.credentials.concat(types.credentials);
+
+		// Copy over all icons and set `iconUrl` for the frontend
```

Comment:
We are copying icons from `/nodes-base/src` to `/nodes-base/dist` with Gulp during build, and here we are copying those icons from `/nodes-base/dist` to the root dir cache. Can we copy to both destinations in the same step, either with Gulp or otherwise?

Reply:
the first part happens during the build process for `nodes-base`, and the second one happens when loading `nodes-base` and community nodes that might also have icons.
We need the first step before we don't publish the icons in the `src` folder, and we need these icons in the published package.